staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go

```diff
@@ -146,11 +158,49 @@ func (t ErrorType) String() string {
 		return "Too many"
 	case ErrorTypeInternal:
 		return "Internal error"
+	case TooLongFailCode:
+		return "Invalid value"
+	case MaxItemsFailCode:
+		return "Invalid value"
+	case TooManyPropertiesCode:
+		return "Invalid value"
+	case InvalidTypeCode:
+		return "Invalid value"
 	default:
 		panic(fmt.Sprintf("unrecognized validation error: %q", string(t)))
 	}
 }
 
+// TooLongFail returns a *Error indicating "MaxLength exceed".
+func TooLongFail(field *Path, value interface{}, detail string) *Error {
+	return &Error{TooLongFailCode, field.String(), value, detail}
+}
+
+// MaxItemsFail returns a *Error indicating "MaxItems exceed"
+func MaxItemsFail(field *Path, value interface{}, detail string) *Error {
+	return &Error{MaxItemsFailCode, field.String(), value, detail}
+}
+
+// TooManyProperties returns a *Error indicating "MaxProperties exceed"
+func TooManyProperties(field *Path, value interface{}, detail string) *Error {
+	return &Error{TooManyPropertiesCode, field.String(), value, detail}
+}
+
+// TypeInvalid returns a *Error indicating "type is invalid"
+func TypeInvalid(field *Path, value interface{}, detail string) *Error {
+	return &Error{InvalidTypeCode, field.String(), value, detail}
+}
+
+// IsCelBlockingErr returns if the error is a CEL blocking error or not
+func (t ErrorType) IsCelBlockingErr() bool {
```

Comment:
make this a helper method in the cel validation package that takes an error type, this package probably shouldn't know about cel

Reply:
could even inline this logic into `hasBlockingErr`